src/composables/useBarDrag.ts

```diff
@@ -1,108 +1,123 @@
-    if (barElement && barContainer) {
-      const barWidth = barElement.getBoundingClientRect().width
-      const xStart = (e.clientX - barContainer.left - cursorOffsetX)
-      const xEnd = xStart + barWidth
-      if (isOutOfRange(xStart, xEnd)) {
-        return
-      }
-      bar.value[barStart.value] = mapPositionToTime(xStart)
-      bar.value[barEnd.value] = mapPositionToTime(xEnd)
-      onDrag(e, bar.value)
+    return { barElement, barContainer }
+  }
+
+  const drag = (e: MouseEvent) => {
+    const { barElement, barContainer } = getBarElements()
+    if (!barElement || !barContainer) {
+      return
     }
+
+    const barWidth = barElement.getBoundingClientRect().width
+    const xStart = e.clientX - barContainer.left - cursorOffsetX
+    const xEnd = xStart + barWidth
+    if (isOutOfRange(xStart, xEnd)) {
+      return
+    }
+    bar.value[barStart.value] = mapPositionToTime(xStart)
+    bar.value[barEnd.value] = mapPositionToTime(xEnd)
+    onDrag(e, bar.value)
   }
 
   const dragByLeftHandle = (e: MouseEvent) => {
-    const barElement = document.getElementById(bar.value.ganttBarConfig.id)
-    const barContainer = barElement?.closest(".g-gantt-row-bars-container")?.getBoundingClientRect()
-    if (barElement && barContainer) {
-      const xStart = e.clientX - barContainer.left
-      const newBarStart = mapPositionToTime(xStart)
-      if (toDayjs(newBarStart).isSameOrAfter(toDayjs(bar.value, "end"))) {
-        return
-      }
-      bar.value[barStart.value] = newBarStart
-      onDrag(e, bar.value)
+    const { barElement, barContainer } = getBarElements()
+    if (!barElement || !barContainer) {
+      return
+    }
+
+    const xStart = e.clientX - barContainer.left
+    const newBarStart = mapPositionToTime(xStart)
+    if (toDayjs(newBarStart).isSameOrAfter(toDayjs(bar.value, "end"))) {
+      return
     }
+    bar.value[barStart.value] = newBarStart
+    onDrag(e, bar.value)
   }
 
   const dragByRightHandle = (e: MouseEvent) => {
-    const barElement = document.getElementById(bar.value.ganttBarConfig.id)
-    const barContainer = barElement?.closest(".g-gantt-row-bars-container")?.getBoundingClientRect()
-    if (barElement && barContainer) {
-      const xEnd = e.clientX - barContainer.left
-      const newBarEnd = mapPositionToTime(xEnd)
-      if (toDayjs(newBarEnd).isSameOrBefore(toDayjs(bar.value, "start"))) {
-        return
-      }
-      bar.value[barEnd.value] = newBarEnd
-      onDrag(e, bar.value)
+    const { barElement, barContainer } = getBarElements()
+    if (!barElement || !barContainer) {
+      return
     }
+
+    const xEnd = e.clientX - barContainer.left
+    const newBarEnd = mapPositionToTime(xEnd)
+    if (toDayjs(newBarEnd).isSameOrBefore(toDayjs(bar.value, "start"))) {
+      return
+    }
+    bar.value[barEnd.value] = newBarEnd
+    onDrag(e, bar.value)
   }
 
   const isOutOfRange = (xStart?: number, xEnd?: number) => {
-    if (!pushOnOverlap) {
+    if (!pushOnOverlap.value) {
```

Comment:
I was convinced that I have fixed this already. Maybe I haven't pushed it to the master branch. In any case, thanks!

Reply:
I only found it with TypeScript if I remember correctly :)